processmanager/process_impl.go

```diff
@@ -12,57 +12,92 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package process
+package processmanager
 
 import (
+	"io"
 	"os"
+	"os/exec"
 	"strings"
 	"syscall"
 	"time"
 
-	"github.com/ligato/cn-infra/process/status"
+	"github.com/ligato/cn-infra/processmanager/status"
 	"github.com/pkg/errors"
 )
 
 // Marked defines that the process should be always restarted
 const infiniteRestarts = -1
 
-func (p *Process) startProcess() (*os.Process, error) {
-	wd, err := os.Getwd()
+func (p *Process) startProcess() (cmd *exec.Cmd, err error) {
+	// set directory
+	cmd = &exec.Cmd{Path: p.cmd}
+	cmd.Dir, err = os.Getwd()
 	if err != nil {
 		return nil, errors.Errorf("failed to get rooted path name for: %v", err)
 	}
-	var attr = os.ProcAttr{
-		Dir:   wd,
-		Env:   os.Environ(),
-		Files: []*os.File{os.Stdin, nil, nil},
-	}
-	// Syscall if process should be detached from parent
-	if p.options != nil && p.options.detach {
-		attr.Sys = &syscall.SysProcAttr{
-			Setpgid: true,
-			Pgid:    0,
+
+	// if options are set, adjust command attributes, otherwise set last required fields to prepare the command
+	if p.options == nil {
+		cmd.Args = append([]string{p.cmd})
+		cmd.SysProcAttr = &syscall.SysProcAttr{
+			Pdeathsig: syscall.SIGKILL,
```

Comment:
It might more convenient to define this signal as global variable, like: `DefaultPDeathSignal`.

Reply:
That's the only `Pdeathsig` signal used in PM plugin. I'm not a fan of global variables, what's the advantage here? The signal itself is already a constant.

Reply:
To allow users to change it to some other type of signal and not be forced to use SIGKILL.

The `Pdeathsig` does not seem significant enough to introduce special option for it, but to allow users to change it, the quickest way is to define global variable for it.

I definitely agree with you that global variables can sometimes be evil, but use of globals is considered bad practice mostly for uses like: storing state or modifying it from many other packages, which can create mutual dependencies and drastically increase complexity.

However there are various other cases where using global variable is adequate. For example, notice the [stdlib](https://golang.org/pkg/) and how some of the packages utilize global variables:

- [net/http](https://golang.org/pkg/net/http/#pkg-variables)
- [flag](https://golang.org/pkg/flag/#pkg-variables)
- [os](https://golang.org/pkg/os/#pkg-variables)

I personally think that using global variables for default values (os) is very clean way to allow users to modify it if needed. Or using globals for default instances (net/http) to allow easier access to instances with sane defaults.